azure-toolkit-libs/azure-toolkit-mysql-lib/src/main/java/com/microsoft/azure/toolkit/lib/mysql/service/AbstractMySqlFirewallRuleCreator.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+package com.microsoft.azure.toolkit.lib.mysql.service;
+
+import com.microsoft.azure.toolkit.lib.common.task.ICommittable;
+import lombok.Getter;
+
+@Getter
+public abstract class AbstractMySqlFirewallRuleCreator implements ICommittable<MySqlFirewallRule> {
+    private String name;
+    private String startIpAddress;
+    private String endIpAddress;
+
+    public AbstractMySqlFirewallRuleCreator withName(String name) {
```

Comment:
Do we need a creator for firewall rules? IMO, firewall rules are more likely the attributes of mysql, like the app settings for app service

Reply:
I prefer to do it like this code.

Reply:
In the future, we may need to use this logic to  create firewall
see `az mysql server firewall-rule create`